src/MoBi.Presentation/Presenter/EditTableFormulaWithXArgumentFormulaPresenter.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using MoBi.Assets;
+using MoBi.Core.Domain.Services;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Mappers;
+using MoBi.Presentation.Views;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Services;
+using OSPSuite.Presentation.Core;
+
+namespace MoBi.Presentation.Presenter
+{
+   public interface IEditTableFormulaWithXArgumentFormulaPresenter : IEditTypedFormulaPresenter
+   {
+      void SetXArgumentFormulaPath();
+      void SetTableObjectPath();
+   }
+
+   public class EditTableFormulaWithXArgumentFormulaPresenter :
+      EditTypedFormulaPresenter<IEditTableFormulaWithXArgumentFormulaView, IEditTableFormulaWithXArgumentFormulaPresenter, TableFormulaWithXArgument>,
+      IEditTableFormulaWithXArgumentFormulaPresenter
+   {
+      private readonly ITableFormulaWithXArgumentToTableFormulaWithXArgumentDTOMapper _mapper;
+      private TableFormulaWithXArgumentDTO _tableFormulaWithXArgumentDTO;
+      private readonly IMoBiFormulaTask _moBiFormulaTask;
+      private readonly IApplicationController _applicationController;
+      private readonly ISelectReferencePresenterFactory _selectReferencePresenterFactory;
+
+      public EditTableFormulaWithXArgumentFormulaPresenter(
+         IEditTableFormulaWithXArgumentFormulaView view,
+         ITableFormulaWithXArgumentToTableFormulaWithXArgumentDTOMapper mapper,
+         IMoBiFormulaTask moBiFormulaTask,
+         IDisplayUnitRetriever displayUnitRetriever,
+         IApplicationController applicationController,
+         ISelectReferencePresenterFactory selectReferencePresenterFactory) : base(view, displayUnitRetriever)
+      {
+         _mapper = mapper;
+         _moBiFormulaTask = moBiFormulaTask;
+         _applicationController = applicationController;
+         _selectReferencePresenterFactory = selectReferencePresenterFactory;
+      }
+
+      public override void Edit(TableFormulaWithXArgument tableFormulaWithXArgument)
+      {
+         _formula = tableFormulaWithXArgument;
+         bindToView();
+      }
+
+      private void bindToView()
+      {
+         _tableFormulaWithXArgumentDTO = _mapper.MapFrom(_formula);
+         _view.BindTo(_tableFormulaWithXArgumentDTO);
+      }
+
+      public void SetXArgumentFormulaPath()
+      {
+         var path = selectFormulaUsablePath(isValidXArgumentObject, AppConstants.Captions.OffsetObjectPath);
+         if (path == null) return;
+
+         AddCommand(_moBiFormulaTask.ChangeXArgumentObject(_formula, path, BuildingBlock));
+         bindToView();
```

Comment:
Code is very very similar to the one before. Yet with some small variation. I could make it work somehow into one but because I know that we will deprecate then remove it soon, I can leave with it